diff --git a/libraries/standard/mqtt/src/iot_mqtt_network.c b/libraries/standard/mqtt/src/iot_mqtt_network.c
index f39eb98..805eca7 100644
--- a/libraries/standard/mqtt/src/iot_mqtt_network.c
+++ b/libraries/standard/mqtt/src/iot_mqtt_network.c
@@ -256,6 +256,12 @@ static IotMqttError_t _allocateAndReceivePacket( IotNetworkConnection_t pNetwork
     IotMqtt_Assert( pMqttConnection != NULL );
     IotMqtt_Assert( pIncomingPacket != NULL );

+#ifdef CBMC
+#ifdef REMAINING_LENGTH_BOUND
+    __CPROVER_assume( pIncomingPacket->remainingLength < REMAINING_LENGTH_BOUND );
+#endif
+#endif
+
     /* Allocate a buffer for the remaining data and read the data. */
     if( pIncomingPacket->remainingLength > 0U )
     {
